src/_utils/crm-web-api.ts

```diff
@@ -500,24 +536,28 @@ export const CRMWebAPI = {
     };
 
       return new Promise((resolve, reject) => {
-          request.post(options, (error, response, body) => {
-            if(body){
-              try{
-                const jsonBody = JSON.parse(body);
-                if(jsonBody.error){
-                  reject(this.parseErrorMessage(jsonBody));
-                }
-                else{
-                  resolve();
-                }
-              }catch(error){
-                reject(body);
+        request.post(options, (error, response, body) => {
+          if (response.statusCode === 403) {
+            resolve("Forbidden.");
+          }
+          if (response.statusCode === 204) {
+            resolve("Uploaded document successfully.")
+          }
+          // If response body exists, 
+          // allow CRM error message to bubble up.
+          if (body) {
+            try {
+              const jsonBody = JSON.parse(body);
+              const parsedErrorMessage = this.parseErrorMessage(jsonBody);
+              if (parsedErrorMessage) {
+                // Nest was throwing server Errors on Promise.reject()?
+                resolve(parsedErrorMessage);
               }
+            } catch(error) {
+              reject(body);
```

Comment:
I vaguely recall an issue with looking up properties on `response` here. What if there is an error response? Is `response` still an object? If not, you'll get an error.

Reply:
Hmm, it seemed when there's an error `response` is still an object, but either `statusCode` will be 403 with no body, or `body` will have an error value.

Reply:
Okidoke, just wanna make sure!